contracts/connectors/loantoken/modules/beaconLogicLM/LoanTokenLogicLM.sol

```diff
@@ -0,0 +1,110 @@
+	 * Every new public function, the signature needs to be included in this function.
+	 *
+	 * @dev This function will return the list of function signature in this contract that are available for public call
+	 * Then this function will be called by LoanTokenLogicBeacon, and the function signatures will be registred in LoanTokenLogicBeacon.
+	 * @dev To save the gas we can just directly return the list of function signature from this pure function.
+	 * The other workaround (fancy way) is we can create a storage for the list of the function signature, and then we can store each function signature to that storage from the constructor.
+	 * Then, in this function we just need to return that storage variable.
+	 *
+	 * @return The list of function signatures (bytes4[])
+	 */
+	function getListFunctionSignatures() external pure returns (bytes4[] memory functionSignatures, bytes32 moduleName) {
+		bytes4[] memory res = new bytes4[](35);
+
+		// Loan Token Logic Standard
+		res[0] = bytes4(keccak256("mint(address,uint256)"));
+		res[1] = bytes4(keccak256("burn(address,uint256)"));
+		res[2] = this.borrow.selector;
+		res[3] = this.marginTrade.selector;
+		res[4] = this.marginTradeAffiliate.selector;
+		res[5] = this.transfer.selector;
+		res[6] = this.transferFrom.selector;
+		res[7] = this.profitOf.selector;
+		res[8] = this.tokenPrice.selector;
+		res[9] = this.checkpointPrice.selector;
+		res[10] = this.marketLiquidity.selector;
+		res[11] = this.avgBorrowInterestRate.selector;
+		res[12] = this.borrowInterestRate.selector;
+		res[13] = this.nextBorrowInterestRate.selector;
+		res[14] = this.supplyInterestRate.selector;
+		res[15] = this.nextSupplyInterestRate.selector;
+		res[16] = this.totalSupplyInterestRate.selector;
+		res[17] = this.totalAssetBorrow.selector;
+		res[18] = this.totalAssetSupply.selector;
+		res[19] = this.getMaxEscrowAmount.selector;
+		res[20] = this.assetBalanceOf.selector;
+		res[21] = this.getEstimatedMarginDetails.selector;
+		res[22] = this.getDepositAmountForBorrow.selector;
+		res[23] = this.getBorrowAmountForDeposit.selector;
+		res[24] = this.checkPriceDivergence.selector;
+		res[25] = this.checkPause.selector;
+		res[26] = this.setLiquidityMiningAddress.selector;
+		res[27] = this.calculateSupplyInterestRate.selector;
+
+		// Loan Token LM
+		/**
+		 * @notice BE CAREFUL,
+		 * LoanTokenLogicStandard also has mint & burn function (overloading).
+		 * You need to compute the function signature manually --> bytes4(keccak256("mint(address,uint256,bool)"))
+		 */
+		res[28] = bytes4(keccak256("mint(address,uint256,bool)"));
+		res[29] = bytes4(keccak256("burn(address,uint256,bool)"));
+
+		// Advanced Token
+		res[30] = this.approve.selector;
+
+		// Advanced Token Storage
+		res[31] = this.totalSupply.selector;
+		res[32] = this.balanceOf.selector;
+		res[33] = this.allowance.selector;
+
+		// Loan Token Logic Storage Additional Variable
+		res[34] = bytes4(keccak256("liquidityMiningAddress()"));
```

Comment:
`res[34] = bytes4(keccak256("liquidityMiningAddress()"));`
i don't think it will work because we already have storage deployed and only the new code that is aware of it will be able to use it, so need to add getter
to cover it with tests we need to deploy the original storage version

Reply:
Good point.
I have created the new separate getter for newly added storage